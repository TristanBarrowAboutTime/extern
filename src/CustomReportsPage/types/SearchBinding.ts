export type SearchBinding = {
    value: string,
    onChange: (event: {target: {value: string}}) => void
} 