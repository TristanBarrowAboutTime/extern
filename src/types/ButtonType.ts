export enum ButtonType {
    SMALL_NORMAL = 'SMALL_NORMAL',
    SMALL_GREEN = 'SMALL_GREEN',
    SMALL_BARE = 'SMALL_BARE',
    SMALL_RED = 'SMALL_RED',
    NORMAL = 'NORMAL',
    GREEN = 'GREEN',
    BARE = 'BARE',
    RED = 'RED',
}