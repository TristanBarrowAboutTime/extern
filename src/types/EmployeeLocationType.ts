export type EmployeeLocationType = {
    code: number
    name: string
    address: string
}