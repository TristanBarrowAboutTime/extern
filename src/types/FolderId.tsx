export type FolderId = { 
    id: number, 
    name: string, 
    editable: boolean 
};