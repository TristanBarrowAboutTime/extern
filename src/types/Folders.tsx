export type Folders = { 
    [index: string]: {
        name: string, 
        editable: boolean
    }
};