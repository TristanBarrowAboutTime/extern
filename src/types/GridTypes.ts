export enum GridType {
    NONE,
    NORMAL,
} 

export class Report {}

export class NormalGridRowData extends Report {
    constructor(reportName: string, description: string, lastRunDate: string) {
        super();
        this.reportName = reportName; 
        this.description = description; 
        this.lastRunDate = lastRunDate; 
    }
    reportName: string
    description: string
    lastRunDate: string
}


