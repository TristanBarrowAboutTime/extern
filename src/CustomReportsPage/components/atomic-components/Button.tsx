import * as React from 'react';
import { FunctionComponent } from 'react';
import styled from 'styled-components';
import { ButtonType } from '../../types/ButtonType';
import { useButton } from '../../hooks/component-hooks/atomic-components/useButton';

type StyledButtonProps = {
    backgroundColor: string
    shadow: string
    padding: string
    cursor: string
    color: string
}

const StyledButton = styled.div`
    border: 0;
    border-radius: 4px;
    white-space: nowrap;
    background-color: ${(props: StyledButtonProps) => props.backgroundColor};
    color: ${(props: StyledButtonProps) => props.color};
    box-shadow: ${(props: StyledButtonProps) => props.shadow};
    padding: ${(props: StyledButtonProps) => props.padding};
    :hover {
        cursor: ${(props: StyledButtonProps) => props.cursor}
    }
    :focus {
        border: 0;
    }
`;

type ButtonProps = {
    buttonType?: ButtonType,
    text: string,
    styles?: object,
    onClick: () => void,
    disabled?: boolean,
}

const Button: FunctionComponent<ButtonProps> = ({
    buttonType = ButtonType.NORMAL, 
    text,
    onClick,
    styles = {},
    disabled = false
}: ButtonProps) => {
    const binding = useButton({buttonType, disabled, onClick});

    return (
        <StyledButton
            style={styles}
            onClick={binding.click}
            backgroundColor={binding.style.backgroundColor}
            padding={binding.style.padding}
            color={binding.style.color}
            shadow={binding.style.shadow}
            cursor={binding.style.cursor}
        >
            {text}
        </StyledButton>
    );
}

export default Button;