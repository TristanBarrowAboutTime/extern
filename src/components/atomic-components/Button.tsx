import React, { FunctionComponent } from 'react';
import styled from 'styled-components';
import Styles from '../../style/Styles';
import { ButtonType } from '../../types/ButtonType';
import { useButton } from '../../hooks/component-hooks/atomic-components/useButton';

/**
 * This will probably need to be turned into react-native at some point.
 */

type StyledButtonProps = {
    backgroundColor: string
    shadow: string
    padding: string
    cursor: string
    color: string
}

// eventually this implementation should be changed to something more sustainable
const StyledButton = styled.button`
    border: 0;
    border-radius: 4px;
    background-color: ${(props: StyledButtonProps) => props.backgroundColor};
    color: ${(props: StyledButtonProps) => props.color};
    box-shadow: ${(props: StyledButtonProps) => props.shadow};
    padding: ${(props: StyledButtonProps) => props.padding};
    :hover {
        cursor: ${(props: StyledButtonProps) => props.cursor}
    }
    :focus {
        border: 0;
    }
`;

type ButtonProps = {
    buttonType?: ButtonType,
    text: string,
    styles?: object,
    onClick: () => void,
    disabled?: boolean,
}

export const Button: FunctionComponent<ButtonProps> = ({
    buttonType = ButtonType.NORMAL, 
    text,
    onClick,
    styles = {},
    disabled = false
}: ButtonProps) => {
    const binding = useButton({buttonType, disabled});

    return (
        <StyledButton
            style={styles}
            onClick={onClick}
            disabled={disabled}
            backgroundColor={binding.backgroundColor}
            padding={binding.padding}
            color={binding.color}
            shadow={binding.shadow}
            cursor={binding.cursor}
        >
            {text}
        </StyledButton>
    );
}


export default Button;