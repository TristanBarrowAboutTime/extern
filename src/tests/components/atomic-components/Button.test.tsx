import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../../components/atomic-components/Button';
import { ButtonType } from '../../../types/ButtonType';
import Style from '../../../style/Styles';
import 'jest-styled-components';
import { unmountComponentAtNode } from 'react-dom';

const style = {
    bgColor: {
        normal: Style.color.gray.xx_dark,
        primary: Style.color.green,
        secondary: Style.color.red,
        bare: Style.color.transparent,
        disabled: Style.color.gray.medium,
    },
    color: {
        normal: Style.color.white,
        primary: Style.color.white,
        secondary: Style.color.white,
        bare: Style.color.gray.xx_dark,
        disabled: Style.color.white,
        bareDisabled: Style.color.gray.dark
    },
    shadow: {
        raised: Style.button.shadow,
        flat: '0',
    },
    size: {
        small: Style.button.paddingSmall,
        normal: Style.button.paddingNormal,
    }

}

const styleNames = {
    bgColor: 'background-color',
    color: 'color',
    padding: 'padding',
    boxShadow: 'box-shadow',
}


let container: Element | DocumentFragment | null  = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});


afterEach(() => {
    unmountComponentAtNode(container as Element | DocumentFragment);
    container = container as Element;
    if (container !== null) container.remove();
    container = null;

});

describe('Button', () => {
    it('gets Clicked when enabled', () => {
        const mockFn = jest.fn();
        const component = renderer.create(
            <Button text={'Get Clicked'} onClick={mockFn} />
        );
        component.root.findAllByType('button')[0].props.onClick();
        expect(mockFn).toHaveBeenCalled();
    });
    
    // done simplest possible
    it('is styled correctly when type is normal and endabled', () => {
        const component = renderer.create(
            <Button text={'Button'} onClick={() => {}}/>
        ).toJSON();
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.normal);
        expect(component).toHaveStyleRule(styleNames.color, style.color.normal);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.raised);
    });

    // done normal enabled
    it('is styled correctly when type is normal and endabled', () => {
        const component = renderer.create(
            <Button 
                buttonType={ButtonType.NORMAL} 
                text={'Button'}
                onClick={() => {}}
            />
        ).toJSON();
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.normal);
        expect(component).toHaveStyleRule(styleNames.color, style.color.normal);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.raised);
    });

    // done small-normal enabled
    it('is styled correctly when type is small-normal and enabled', () => {
        const component = renderer.create(
            <Button 
                buttonType={ButtonType.SMALL_NORMAL} 
                text={'Button'}
                onClick={() => {}}
            />
        ).toJSON();
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.normal);
        expect(component).toHaveStyleRule(styleNames.color, style.color.normal);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.raised);
    });
    
    // normal disabled
    it('is styled correctly when type is normal and disabled', () => {
        const component = renderer.create(
            <Button 
                buttonType={ButtonType.NORMAL} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.disabled);
        expect(component).toHaveStyleRule(styleNames.color, style.color.disabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });

    // small-normal disabled
    it('is styled correctly when type is small-normal and disabled', () => {
        const component = renderer.create(
            <Button 
                buttonType={ButtonType.SMALL_NORMAL} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.disabled);
        expect(component).toHaveStyleRule(styleNames.color, style.color.disabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });

    // green enabled
    it('is styled correctly when type is green and enabled', () => {
        const component = renderer.create(
            <Button 
                buttonType={ButtonType.GREEN} 
                text={'Button'} 
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.primary);
        expect(component).toHaveStyleRule(styleNames.color, style.color.primary);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.raised);
    });

    // small-green enabled
    it('is styled correctly when type is small-green and enabled', () => {
        const component = renderer.create(
             <Button 
                buttonType={ButtonType.SMALL_GREEN} 
                text={'Button'} 
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.primary);
        expect(component).toHaveStyleRule(styleNames.color, style.color.primary);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.raised);
    });

    // green disabled
    it('is styled correctly when type is green and disabled', () => {
        const component = renderer.create(
             <Button 
                buttonType={ButtonType.GREEN} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.disabled);
        expect(component).toHaveStyleRule(styleNames.color, style.color.disabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });

    // small-green disabled
    it('is styled correctly when type is small-green disabled', () => {
        const component = renderer.create(
              <Button 
                buttonType={ButtonType.SMALL_GREEN} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.disabled);
        expect(component).toHaveStyleRule(styleNames.color, style.color.disabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });

    // red enabled
    it('is styled correctly when type is red and endabled', () => {
        const component = renderer.create(
             <Button 
                buttonType={ButtonType.RED} 
                text={'Button'} 
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.secondary);
        expect(component).toHaveStyleRule(styleNames.color, style.color.secondary);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.raised);
    });

    // small-red enabled
    it('is styled correctly when type is small-red and enabled', () => {
        const component = renderer.create(
             <Button 
                buttonType={ButtonType.SMALL_RED} 
                text={'Button'} 
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.secondary);
        expect(component).toHaveStyleRule(styleNames.color, style.color.secondary);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.raised);
    });

    // red disabled
    it('is styled correctly when type is red and disabled', () => {
        const component = renderer.create(
              <Button 
                buttonType={ButtonType.RED} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.disabled);
        expect(component).toHaveStyleRule(styleNames.color, style.color.disabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });

    // small-red disabled
    it('is styled correctly when type is small-red and disabled', () => {
        const component = renderer.create(
             <Button 
                buttonType={ButtonType.SMALL_RED} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.disabled);
        expect(component).toHaveStyleRule(styleNames.color, style.color.disabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });
    
    // bare enabled
    it('is styled correctly when type is bare and enabled', () => {
        const component = renderer.create(
             <Button 
                buttonType={ButtonType.BARE} 
                text={'Button'} 
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.bare);
        expect(component).toHaveStyleRule(styleNames.color, style.color.bare);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });

    // small-bare enabled
    it('is styled correctly when type is small-bare and enabled', () => {
        const component = renderer.create(
             <Button 
                buttonType={ButtonType.SMALL_BARE} 
                text={'Button'} 
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.bare);
        expect(component).toHaveStyleRule(styleNames.color, style.color.bare);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });
    
    // bare disabled
    it('is styled correctly when type is bare and disabled', () => {
        const component = renderer.create(
              <Button 
                buttonType={ButtonType.BARE} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.bare);
        expect(component).toHaveStyleRule(styleNames.color, style.color.bareDisabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.normal);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });

    // small-bare disabled
    it('is styled correctly when type is small-bare and disabled', () => {
        const component = renderer.create(
               <Button 
                buttonType={ButtonType.SMALL_BARE} 
                text={'Button'} 
                disabled
                onClick={() => {}}
            />
        ).toJSON()
        expect(component).toMatchSnapshot();
        expect(component).toHaveStyleRule(styleNames.bgColor, style.bgColor.bare);
        expect(component).toHaveStyleRule(styleNames.color, style.color.bareDisabled);
        expect(component).toHaveStyleRule(styleNames.padding, style.size.small);
        expect(component).toHaveStyleRule(styleNames.boxShadow, style.shadow.flat);
    });
});
