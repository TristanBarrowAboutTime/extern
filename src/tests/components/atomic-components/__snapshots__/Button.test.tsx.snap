// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is styled correctly when type is bare and disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: rgba(255,255,255,0);
  color: #777777;
  box-shadow: 0;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#777777"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is bare and enabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: rgba(255,255,255,0);
  color: #333333;
  box-shadow: 0;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#333333"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is green and disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #A7AFB2;
  color: #FFFFFF;
  box-shadow: 0;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is green and enabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #85B554;
  color: #FFFFFF;
  box-shadow: 0 2px 4px #cccccc;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is normal and disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #A7AFB2;
  color: #FFFFFF;
  box-shadow: 0;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is normal and endabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #333333;
  color: #FFFFFF;
  box-shadow: 0 2px 4px #cccccc;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is normal and endabled 2`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #333333;
  color: #FFFFFF;
  box-shadow: 0 2px 4px #cccccc;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is red and disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #A7AFB2;
  color: #FFFFFF;
  box-shadow: 0;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is red and endabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #9B3E38;
  color: #FFFFFF;
  box-shadow: 0 2px 4px #cccccc;
  padding: 10px 12px 8px 12px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-bare and disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: rgba(255,255,255,0);
  color: #777777;
  box-shadow: 0;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#777777"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-bare and enabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: rgba(255,255,255,0);
  color: #333333;
  box-shadow: 0;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#333333"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-green and enabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #85B554;
  color: #FFFFFF;
  box-shadow: 0 2px 4px #cccccc;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-green disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #A7AFB2;
  color: #FFFFFF;
  box-shadow: 0;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-normal and disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #A7AFB2;
  color: #FFFFFF;
  box-shadow: 0;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-normal and enabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #333333;
  color: #FFFFFF;
  box-shadow: 0 2px 4px #cccccc;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-red and disabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #A7AFB2;
  color: #FFFFFF;
  box-shadow: 0;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: default;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="default"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;

exports[`Button is styled correctly when type is small-red and enabled 1`] = `
.c0 {
  border: 0;
  border-radius: 4px;
  background-color: #9B3E38;
  color: #FFFFFF;
  box-shadow: 0 2px 4px #cccccc;
  padding: 6px 17px 4px 17px;
}

.c0:hover {
  cursor: cursor;
}

.c0:focus {
  border: 0;
}

<button
  className="c0"
  color="#FFFFFF"
  cursor="cursor"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
>
  Button
</button>
`;
